smart-contracts/contracts/mixins/MixinKeys.sol

```diff
@@ -31,11 +31,21 @@ contract MixinKeys is
 
   event KeyManagerChanged(uint indexed _tokenId, address indexed _newManager);
 
-  // Keys
-  // Each owner can have at most exactly one key
-  // return 0 values when missing a key
+  // DEPREC: dont use
   mapping (address => Key) internal keyByOwner;
 
+  // store all keys: tokenId => token
+  mapping(uint256 => Key) private _keys;
+  
+  // store ownership: owner => list of tokenIds
+  mapping(address => mapping(uint256 => uint256)) private _ownedKeyIds;
+  
+  // store indexes: owner => list of tokenIds
+  mapping(uint256 => uint256) private _ownedKeysIndex;
+
+  // Mapping owner address to token count
+  mapping(address => uint256) private _balances;
```

Comment:
Wow. is there no way to know the size of a mapping?

Reply:
nope, not in solidity